src/FSharp.Analyzers/UnionCaseAnalyzer.fs

```diff
@@ -56,9 +56,18 @@ let findAllShadowingCases
                 "Error"
             ]
 
-    let handleCase (SynUnionCase (ident = (SynIdent (ident, _)))) =
+    let handleCase (typeKeywordRange : Range) (SynUnionCase (ident = (SynIdent (ident, _)))) =
         if (namesToWarnAbount |> Set.contains ident.idText) then
-            collector.Add ident.idRange
+            let indent = String.replicate typeKeywordRange.StartColumn " "
+
+            let fix =
+                {
+                    FromText = ""
+                    FromRange = typeKeywordRange.StartRange
+                    ToText = $"[<RequireQualifiedAccess>]\n{indent}"
```

Comment:
This can't go wrong with xml document or existing attribute being present on the type right?
And about the `and` keyword?

Reply:
> This can't go wrong with xml document or existing attribute being present on the type right?
No, that's safe.

And about the `and` keyword?
Oh, yeah sorry. Good catch.